File: backend/internal/health/health.go
// Package health は Go Cron Worker の稼働確認用ペイロードを提供する。
package health

import "encoding/json"

// Payload は稼働確認レスポンスの構造。
type Payload struct {
	Status  string `json:"status"`
	Service string `json:"service"`
}

// New は ok 状態のペイロードを返す。
func New() Payload {
	return Payload{Status: "ok", Service: "yuruwollet-cron"}
}

// PayloadJSON は ok ペイロードを JSON 文字列で返す。
func PayloadJSON() string {
	b, _ := json.Marshal(New())
	return string(b)
}

File: frontend/functions/api/health.ts
export interface HealthPayload {
  status: 'ok';
  service: 'yuruwollet';
}

export function buildHealth(): HealthPayload {
  return { status: 'ok', service: 'yuruwollet' };
}

/**
 * Cloudflare Pages Function: `GET /api/health`
 * Phase 2 で @cloudflare/workers-types による厳密な型付け + Access 検証に発展させる。
 */
export const onRequest = (): Response =>
  new Response(JSON.stringify(buildHealth()), {
    headers: { 'content-type': 'application/json; charset=utf-8' },
  });
